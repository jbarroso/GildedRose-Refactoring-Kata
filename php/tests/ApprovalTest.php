<?php

namespace Tests;

use ApprovalTests\Approvals;
use ApprovalTests\CombinationApprovals;
use GildedRose\GildedRose;
use GildedRose\Item;
use PHPUnit\Framework\TestCase;

class ApprovalTest extends TestCase
{
    public function testList(): void
    {
        $items = [
            new Item('+5 Dexterity Vest', 10, 20),
            new Item('Aged Brie', 2, 0),
            new Item('Elixir of the Mongoose', 5, 7),
            new Item('Sulfuras, Hand of Ragnaros', 0, 80),
            new Item('Sulfuras, Hand of Ragnaros', -1, 80),
            new Item('Backstage passes to a TAFKAL80ETC concert', 15, 20),
            new Item('Backstage passes to a TAFKAL80ETC concert', 10, 49),
            new Item('Backstage passes to a TAFKAL80ETC concert', 5, 49),
            // this conjured item does not work properly yet
            //new Item('Conjured Mana Cake', 3, 6),
        ];

        $app = new GildedRose($items);

        $days = 10;

        for ($i = 0; $i < $days; $i++) {
            $app->updateQuality();
        }
        Approvals::verifyList($items);
    }

    public function testCombinations(): void
    {
        $names = [
            '+5 Dexterity Vest',
            'Aged Brie',
            'Elixir of the Mongoose',
            'Sulfuras, Hand of Ragnaros',
            'Backstage passes to a TAFKAL80ETC concert',
            //'Conjured Mana Cake'
        ];
        $sellIns = range(-1, 15);
        $qualities = range(0, 80);

        CombinationApprovals::verifyAllCombinations3(function ($name, $sellIn, $quality) {
            $item = new Item($name, $sellIn, $quality);
            $app = new GildedRose([$item]);
            $app->updateQuality();
            return $item;
        }, $names, $sellIns, $qualities);
    }
}
