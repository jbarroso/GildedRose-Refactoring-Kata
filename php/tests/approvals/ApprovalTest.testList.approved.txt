[0] -> +5 Dexterity Vest, 0, 10
[1] -> Aged Brie, -8, 18
[2] -> Elixir of the Mongoose, -5, 0
[3] -> Sulfuras, Hand of Ragnaros, 0, 80
[4] -> Sulfuras, Hand of Ragnaros, -1, 80
[5] -> Backstage passes to a TAFKAL80ETC concert, 5, 35
[6] -> Backstage passes to a TAFKAL80ETC concert, 0, 50
[7] -> Backstage passes to a TAFKAL80ETC concert, -5, 0
